### PR TITLE
[stable/external-dns] Add support for creating Azure secret

### PR DESCRIPTION
**Pull request by invidian**
#### What this PR does / why we need it:

This commit allows Azure users to only use chart values when deploying
external-dns with azure provider.

Similar to AWS credentials, it adds helper template for azure.json file,
appropriate new values and logic.

azure.secretName is always prefered over new values to keep backward
compatibility. if azure.secretName is not set and new values are not
satisfied, template will fallback to hosts /etc/kubernetes/azure.json
file.

Also fixes table formatting and typos in README.md

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/external-dns/values.yaml [hunk 1]
```diff
@@ -81,10 +81,27 @@ azure:
...
+  subscriptionId: ""
+  ## Azure Application Client ID to use
+  ##
+  aadClientId: ""
+  ## Azure Application Client Secret to use
+  ##
+  aadClientSecret: ""
+  ## If you use Azure MSI, this should be set to true
+  ##
+  useManagedIdentityExtension: false
```
Comment (juan131) [inline stable/external-dns/values.yaml:104]: Please also update the **values-production.yaml** with the same changes.

File: stable/external-dns/templates/secret.yaml [hunk 2]
```diff
@@ -33,4 +33,7 @@ data:
   {{- if .Values.pdns.apiKey }}
   pdns_api_key: {{ .Values.pdns.apiKey | b64enc | quote }}
   {{- end }}
+  {{- if eq .Values.provider "azure" }}
+  azure.json: {{ include "external-dns.azure-credentials" . | b64enc | quote }}
```
Comment (juan131) [inline stable/external-dns/templates/secret.yaml]: Small fix:
```diff
-   azure.json: {{ include "external-dns.azure-credentials" . | b64enc | quote }}
+   azure.json: {{ (include "external-dns.azure-credentials" .) | b64enc | quote }}
```

File: stable/external-dns/templates/deployment.yaml [hunk 3]
```diff
@@ -327,13 +327,16 @@ spec:
       # Azure volume(s)
       {{- if eq .Values.provider "azure" }}
       - name: azure-config-file
-        {{- if (not .Values.azure.secretName) }}
+        {{- if .Values.azure.secretName }}
+        secret:
+          secretName: {{ .Values.azure.secretName }}
+        {{- else if (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId .Values.azure.aadClientId .Values.azure.aadClientSecret (not .Values.azure.useManagedIdentityExtension)) (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId .Values.azure.useManagedIdentityExtension) }}
```
Comment (juan131) [inline stable/external-dns/templates/deployment.yaml]: Wouldn't be easier to just check the mandatory parameters?
```diff
- {{- else if (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId .Values.azure.aadClientId .Values.azure.aadClientSecret (not .Values.azure.useManagedIdentityExtension)) (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId .Values.azure.useManagedIdentityExtension) }}
+ {{- else if (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId }}
```
Then you can add a `{{- define "external-dns.validateValues.azureConfiguration }}` function in the helpers (check how we did it for other "external-dns.validateValues" functions) to ensure the `addClientId` and `addClientSecret` are not set when `useManagedIdentityExtension` is set, and viceversa.

File: stable/external-dns/Chart.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: external-dns
-version: 2.5.5
+version: 2.5.6
```
Comment (juan131) [inline stable/external-dns/Chart.yaml]: Bump minor version

**Review by juan131 — CHANGES_REQUESTED** [review]
Hi @invidian 

Thanks so much for your contribution. Could you please review my comments?

File: stable/external-dns/templates/secret.yaml (hunk 2 above)
Comment (invidian) [reply stable/external-dns/templates/secret.yaml]: Fixed.

File: stable/external-dns/values.yaml (hunk 1 above)
Comment (invidian) [reply stable/external-dns/values.yaml:104]: Right, fixed.

File: stable/external-dns/Chart.yaml (hunk 4 above)
Comment (invidian) [reply stable/external-dns/Chart.yaml]: Changed to 2.6.0 then.

File: stable/external-dns/templates/deployment.yaml (hunk 3 above)
Comment (invidian) [reply stable/external-dns/templates/deployment.yaml]: All right, took me a while to figure out how to do that, but I just pushed the changes :+1: Cool that you can do validation in helm such way, TIL.

**Comment by invidian** [comment]
Any updates on this? Also rebased to resolve merge conflicts.

**Review by juan131 — APPROVED** [review]
wow! it's so impressive you implemented so many validations!!!

Thanks so much for this great contribution and sorry for the late response, we lost this PR. I'm approving your changes

**Comment by juan131** [comment]
Once you fix the conflict with the master branch, I'll trigger the merging process @invidian

**Comment by invidian** [comment]
Conflicts resolved :+1:

**Comment by juan131** [comment]
It happened again. So sorry for this. I tried to fix the conflicts myself but I had no permissions to do so in your fork repo.

**Comment by invidian** [comment]
No worries, rebased again.

**Review by juan131 — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**